python/dolfinx/fem/petsc.py

```diff
@@ -1118,3 +1121,344 @@ def interpolation_matrix(space0: _FunctionSpace, space1: _FunctionSpace) -> PETS
         Interpolation matrix.
     """
     return _interpolation_matrix(space0._cpp_object, space1._cpp_object)
+
+
+class SNESProblem:
+    def __init__(
+        self,
+        F: typing.Union[dolfinx.fem.Form, ufl.form.Form],
+        u: dolfinx.fem.Function,
+        J: typing.Optional[typing.Union[dolfinx.fem.Form, ufl.form.Form]] = None,
+        P: typing.Optional[typing.Union[dolfinx.fem.Form, ufl.form.Form]] = None,
+        bcs: typing.Optional[list[dolfinx.fem.DirichletBC]] = None,
+        form_compiler_options: typing.Optional[dict] = None,
+        jit_options: typing.Optional[dict] = None,
+    ):
+        """Class for constructing the residual and Jacobian constructors for a SNES problem.
+
+        Args:
+            F: The PDE residual F(u, v)
```

Comment:
`F(u, v)` doesn't mean much in isolation. A description of what the class does, and in the argument docs refer to what the argument is the math formulation.